src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -508,6 +522,62 @@ private void ValidateLicenseFile(IEnumerable<IPackageFile> files, LicenseMetadat
             }
         }
 
+        /// <summary>
+        /// Given a list of resolved files,
+        /// determine which file will be used as the icon file and validate its size.
+        /// </summary>
+        /// <param name="files">Files resolved from the file entries in the nuspec</param>
+        /// <param name="iconPath">iconpath found in the .nuspec</param>
+        /// <exception cref="PackagingException">When a validation rule is not met</exception>
+        private void ValidateIconFile(IEnumerable<IPackageFile> files, string iconPath)
```

Comment:
If you make this method internal, you can write tests against it directly and use mock `IPackageFile`, so the tests will execute faster and require less boilerplate code.

Reply:
I made it similar to ValidateLicenseFile(), I will consider making these methods internal, but in another PR.